### PR TITLE
Add createIssueDiscussion endpoint

### PR DESCRIPTION
**Pull request by svendroid**
Allows to create a new comment for an issue by using the DiscussionsApi:
https://docs.gitlab.com/ee/api/discussions.html#create-new-issue-thread

### CODE REVIEW

**Comment by gmessner** [comment]
@svendroid Thanks for the contribution.  To make this complete the following methods should also be added:
```
public Note addIssueThreadNote(Object projectIdOrPath, Integer issueIid,
	    String discussionId, String body, Date createdAt) 
 public Note modifyIssueThreadNote(Object projectIdOrPath, Integer issueIid,
	    String discussionId, Integer noteId, String body)
public void deleteIssueThreadNote(Object projectIdOrPath, Integer issueIid,
	    String discussionId, Integer noteId)
```
The documentation at https://docs.gitlab.com/ee/api/discussions.html#add-note-to-existing-snippet-thread for "Add note to existing issue thread" indicates that a ```note_id``` should be included but I beleive this is incorrect as the call itself creates the note.

**Comment by svendroid** [comment]
I added methods to add, modify & delete notes for a discussion of an issue. The gitlab API also use the term thread for a discussion. I went with Discussion because the rest Endpoint is called discussions and the model inside gitlab4j is also a Discussion.

> The documentation at https://docs.gitlab.com/ee/api/discussions.html#add-note-to-existing-snippet-thread for "Add note to existing issue thread" indicates that a note_id should be included but I beleive this is incorrect as the call itself creates the note.
The link describes the notes for a snippet discussion. Maybe it is a copy paste error? Did you mean https://docs.gitlab.com/ee/api/discussions.html#add-note-to-existing-issue-thread? This describes the notes regarding notes on issue discussions.

**Comment by gmessner** [comment]
@svendroid Yes, I referenced the wrong link, and thanks for adding the new methods.

That being said, please use the names as I provided as that is the pattern this library uses for each of the note types (commit and merge request), it is best to be consistent and more in line with the GitLab API docs.

**Comment by svendroid** [comment]
@gmessner thanks for the feedback. I renamed the methods.

File: src/main/java/org/gitlab4j/api/DiscussionsApi.java [hunk 1]
```diff
@@ -671,4 +671,86 @@ public void deleteMergeRequestThreadNote(Object projectIdOrPath, Integer mergeRe
...
+
+    /**
+     * Deletes an existing thread note of an issue.
+     *
+     * <pre><code>GitLab Endpoint: DELETE /projects/:id/issues/:issue_iid/discussions/:discussion_id/notes/:note_id</code></pre>
+     *
+     * @param projectIdOrPath projectIdOrPath the project in the form of an Integer(ID), String(path), or Project instance
+     * @param issueIid The IID of an issue
+     * @param discussionId the id of discussion
+     * @param noteId the id of the note
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/DiscussionsApi.java]: Need to add the following to the Javadocs:
```
* @throws GitLabApiException if any exception occurs during execution
```

**Comment by gmessner** [comment]
@svendroid Thanks for the quick turnaround.  Just one small change and it is set to go.  If you are not able to get to the change today, I will go ahead and merge and then make the change, because I want to get a release out today.

File: src/main/java/org/gitlab4j/api/DiscussionsApi.java (hunk 1 above)
Comment (svendroid) [reply src/main/java/org/gitlab4j/api/DiscussionsApi.java]: added the javadoc for the exception

**Comment by svendroid** [comment]
@gmessner fixed the javadoc. Should I squash the commits into one? or will this be done automatically when you merge?

**Comment by gmessner** [comment]
@svendroid That was fast.  No need to squash, that will happen when I merge it.

**Comment by gmessner** [comment]
@svendroid GitLab4J-API release 4.14.19 is available with this functionality.

**Pull request merged**